Build failure: pkgsLLVM.screen

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

1. Build `pkgsLLVM.screen`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

N/A

### Relevant log output

```console
screen> Running phase: checkPhase
screen> check flags: SHELL=/nix/store/xgvqbslsa5km8q1bkhlhdr3r5kxblvn6-bash-5.2p37/bin/bash VERBOSE=y check
screen> aarch64-unknown-linux-gnu-clang -c -iquote. -DSCREENENCODINGS='"/nix/store/gpv5zw5mknhy861ypj262xf0xp558q6n-screen-aarch64-unknown-linux-gnu-5.0.0/share/screen/utf8encodings"' -DBUILD_DATE='"2024-08-28 19:08:22"' -g -O2 -Wall -Wextra -std=c17 tests/mallocmock.c -o tests/mallocmock.o
screen> aarch64-unknown-linux-gnu-clang -iquote. -DSCREENENCODINGS='"/nix/store/gpv5zw5mknhy861ypj262xf0xp558q6n-screen-aarch64-unknown-linux-gnu-5.0.0/share/screen/utf8encodings"' -DBUILD_DATE='"2024-08-28 19:08:22"' -g -O2 -Wall -Wextra -std=c17 tests/test-winmsgbuf.c -o tests/test-winmsgbuf winmsgbuf.o tests/mallocmock.o
screen> aarch64-unknown-linux-gnu-ld: error: undefined symbol: mallocmock_reset
screen> >>> referenced by test-winmsgbuf.c:62 (/build/screen-5.0.0/tests/test-winmsgbuf.c:62)
screen> >>>               /build/test-winmsgbuf-2a4e84.o:(main)
screen> >>> referenced by test-winmsgbuf.c:66 (/build/screen-5.0.0/tests/test-winmsgbuf.c:66)
screen> >>>               /build/test-winmsgbuf-2a4e84.o:(main)
screen> >>> referenced by test-winmsgbuf.c:67 (/build/screen-5.0.0/tests/test-winmsgbuf.c:67)
screen> >>>               /build/test-winmsgbuf-2a4e84.o:(main)
screen> >>> referenced 1 more times
screen> clang: error: linker command failed with exit code 1 (use -v to see invocation)
screen> make: *** [Makefile:77: tests/test-winmsgbuf] Error 1
```

### Additional context

_No response_

### System metadata

 - system: `"aarch64-linux"`
 - host os: `Linux 6.6.67, NixOS, 24.11 (Vicuna), 24.11.20241225.4005c3f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/smgkshm3r00a4rl7gw5ymgfqsd2k573l-source`

### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/misc/screen/default.nix
+++ b/pkgs/tools/misc/screen/default.nix
@@ -24,7 +24,7 @@ stdenv.mkDerivation rec {
   ];
 
   # We need _GNU_SOURCE so that mallocmock_reset() is defined: https://savannah.gnu.org/bugs/?66416
-  NIX_CFLAGS_COMPILE = lib.optionalString (stdenv.cc.isGNU) "-D_GNU_SOURCE=1 -Wno-int-conversion -Wno-incompatible-pointer-types";
+  NIX_CFLAGS_COMPILE = "-D_GNU_SOURCE=1 -Wno-int-conversion -Wno-incompatible-pointer-types";
 
   patches = [
     # GNU Screen 5.0 uses strncpy incorrectly in SendCmdMessage
@@ -43,7 +43,8 @@ stdenv.mkDerivation rec {
   ] ++ lib.optional stdenv.hostPlatform.isDarwin utmp;
 
   # The test suite seems to have some glibc malloc hooks that don't exist/link on macOS
-  doCheck = !stdenv.hostPlatform.isDarwin;
+  # With pkgsLLVM: tests/test-winmsgcond.c:53: assertion 'wmc_end(&wmc, pos + 1, &chg) == pos' failed
+  doCheck = !stdenv.hostPlatform.isDarwin && !stdenv.hostPlatform.useLLVM;
 
   meta = with lib; {
     homepage = "https://www.gnu.org/software/screen/";

